product.py

```diff
@@ -0,0 +1,387 @@
+# The COPYRIGHT file at the top level of this repository contains the full
+# copyright notices and license terms.
+import functools
+from trytond.model import fields, ModelSQL, ModelView
+from trytond.pool import PoolMeta, Pool
+from trytond.pyson import Eval, Bool, Id, Or
+from trytond.transaction import Transaction
+from trytond.tools import grouped_slice
+from trytond.model.exceptions import AccessError
+from trytond.i18n import gettext
+from decimal import Decimal
+
+NON_MEASURABLE = ['service']
+
+def check_no_move(func):
+
+    @functools.wraps(func)
+    def decorator(cls, *args):
+        Product =  Pool().get('product.product')
+        Template =  Pool().get('product.template')
+        ProductPackage = Pool().get('product.product-product.packaging')
+
+        transaction = Transaction()
+        if (transaction.user != 0 and transaction.context.get('_check_access')):
+            actions = iter(args)
+            for records, values in zip(actions, actions):
+                for field, msg in Product._modify_no_move:
+                    products = []
+                    if records and isinstance(records[0], Template):
+                        for record in records:
+                            products += record.products
+                    if records and isinstance(records[0], ProductPackage):
+                        products = [x.product for x in records]
+                    if field in values:
+                        if Product.find_move(products):
+                            raise AccessError(gettext(msg))
+                        break
+        func(cls, *args)
+    return decorator
+
+
+class ProductProductPackaging(ModelSQL, ModelView):
+    "Product - Product Packaging"
+    __name__ = 'product.product-product.packaging'
+
+    production_template = fields.Many2One('production.template',
+        'Production Template', required=True, ondelete='CASCADE',
+            states ={
+                'readonly': Bool(Eval('packaged_product')),
+            },
+            domain=[
+             ['OR', ('packaging', '=', True), ('labeling', '=', True)],
+             ('inputs_products', 'in', Eval('product')),
```

Comment:
Tinc dubtes que funcioni un "in" amb un m2o.

Per exemple un "in" i Eval si es none retorni una llista:

```
('invoice.type', 'in', Eval('invoice_types', [])),
```